### DISCUSSION TITLE
Clicking heading link in markdown doesn't scroll to heading

### DISCUSSION THREAD

**Discussion by m-spitfire** [community/community#46856] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

This is a bug for new "Code search and navigation".

### Steps to reproduce:
- Open a markdown file in new navigation view, e.g. https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md
- Click to a heading in the open file (in above file, one of the names of language servers)
### Expected
File to scroll to the heading
## Actual result
Nothing happens except change of the page URL.

One more additional note: when opening a heading [link directly](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#ansiblels), it goes to the correct place.

**Comment by look** [comment, marked as answer]
Hi @m-spitfire, thank you for the detailed bug report. This is working for me, can you give more details about what browser you are using? A video is helpful, too!

**Reply by m-spitfire** [reply]
Hi @look, sorry for the late response. It is working for me now too, I think I had an extension that was interfering with the expected behavior, but I am not sure. Anyway, sorry for the incorrect issue.

**Comment by spamegg1** [comment]
I have the same issue in April 2024, using Firefox and Opera on Linux (all extensions disabled).
The bug is still present! On both browsers.
After clicking a header link and *not* scrolling to the correct location, if I reload the page with Ctrl+R or F5, then it *does* scroll to the correct location.

Here is an example from one of my repositories. Using relative links in Markdown using headers.

1. Go here: [my repo](https://github.com/spamegg1/abstract-thinking/)
2. Scroll down a bit. Find the part called "Metacognition" and click the link that takes you to `metacog.md`. This link works correctly and takes you to the link https://github.com/spamegg1/abstract-thinking/blob/main/metacog.md In Markdown, this link is coded as: `[Link to the Metacognition section](metacog.md)`
3. At the bottom of `metacog.md` there is a link named "Back to Elements" which is *supposed to* go straight back to the Metacognition section in the main README, in other words, to: https://github.com/spamegg1/abstract-thinking/blob/main/README.md#metacognition In Markdown this link is coded as: `[Back to Elements](README.md#metacognition)`
4. Click on that. **_See the bug._** The page is *at the top of* the main README. It does not properly scroll to the Metacognition section.
5. Now reload the page with F5. This time, it *does* scroll!

Here is a video demonstration:

https://github.com/community/community/assets/4255997/4f65c81c-d733-4be9-b482-7f6a729ddb3f
